File: Maximum Product of Word Lengths.java
public class Solution {
    public int maxProduct(String[] words_) {
        int gmax = 0, size = words_.length;
        int words[] = new int[size];
        for (int i = 0; i < size; i++){
            int sb = 0;
            for (Character c : words_[i].toCharArray()) sb |= 1 << (c-'a');
            words[i] = sb;
        }
        for (int i = 0; i < size-1; i++)
            for (int j = i+1; j < size; j++)
                if ((words[i] & words[j]) == 0)
                    gmax = Math.max(words_[i].length() * words_[j].length(), gmax);
        return gmax;
    }
}

File: Maximum Product of Word Lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words_) {
        size_t gmax = 0;
        vector<unsigned> words;
        for (auto s : words_){
            unsigned sb = 0;
            for (auto c : s) sb |= 1 << (c-'a');
            words.emplace_back(sb);
        }
        for (size_t i = 0; i < words.size()-1; i++)
            for (size_t j = i+1; j < words.size(); j++)
                if (!(words[i] & words[j]))
                    gmax = max(words_[i].size() * words_[j].size(), gmax);
        return gmax;
    }
private:
    bool testEqual(const string& s1, const string& s2){
        bitset<26> sb1;
        bitset<26> sb2;
        for (auto c : s1) sb1.set(c-'a');
        for (auto c : s2) sb2.set(c-'a');
        return !(sb1 & sb2).count();
    }
};